include/envoy/registry/registry.h

```diff
@@ -318,7 +383,53 @@ template <class T, class Base> class RegisterFactory {
     }
   }
 
+  /**
+   * Constructor that registers an instance of the factory with the FactoryRegistry along with
+   * vendor specific version.
+   */
+  RegisterFactory(uint32_t major, uint32_t minor, uint32_t patch,
+                  std::initializer_list<absl::string_view> labels)
+      : RegisterFactory(major, minor, patch, labels, {}) {}
+
+  /**
+   * Constructor that registers an instance of the factory with the FactoryRegistry along with
+   * vendor specific version and deprecated names.
+   */
+  RegisterFactory(uint32_t major, uint32_t minor, uint32_t patch,
+                  std::initializer_list<absl::string_view> labels,
+                  std::initializer_list<absl::string_view> deprecated_names) {
+    auto version = MakeBuildVersion(major, minor, patch, labels);
+    if (!instance_.name().empty()) {
+      FactoryRegistry<Base>::registerFactory(instance_, instance_.name(), version);
+    } else {
+      ASSERT(deprecated_names.size() != 0);
```

Comment:
Nit: `!deprecated_names.empty()`

Reply:
std::initializer_list does not have empty() unfortunately